src/main/java/com/google/sps/data/Card.java

```diff
@@ -107,7 +118,65 @@ public Entity createEntity() {
     card.setProperty("imageBlobKey", this.imageBlobKey);
     card.setProperty("rawText", this.rawText);
     card.setProperty("textTranslated", this.textTranslated);
+    card.setProperty("status", true);
 
     return card;
   }
+
+  public static Card storeCardInDatastore(Card card, DatastoreService datastore, String folderKey) {
+    card.setParentKey(folderKey);
+    Entity cardEntity = card.createEntity();
+    datastore.put(cardEntity);
+
+    card.setCardKey(KeyFactory.keyToString(cardEntity.getKey()));
+
+    return card;
+  }
+
+  public static void deleteCard(Entity card) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    markCardAsDeleted(card);
+    try {
+      datastore.delete(card.getKey());
+    } catch (DatastoreFailureException e) {
+      throw e;
+    }
+  }
+
+  public static void markCardAsDeleted(Entity card) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    card.setProperty("status", false);
+    datastore.put(card);
+  }
+
+  public static void deleteBlob(String blobKey) throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    BlobKey key = new BlobKey(blobKey);
+
+    int retries = 5;
+    while (true) {
+      try {
+        blobstoreService.delete(key);
+        break;
+      } catch (BlobstoreFailureException e) {
+        if (retries == 0) {
+          addBlobstoreDeleteTaskToQueue(key.getKeyString());
+          break;
+        }
+        --retries;
+      }
+    }
+  }
+
+  public static void addBlobstoreDeleteTaskToQueue(String key) {
+    Queue queue = QueueFactory.getDefaultQueue();
```

Comment:
These two methods also don't seem like they belong in Card.

Reply:
done